src/uu/split/src/split.rs

```diff
@@ -40,11 +41,23 @@ static OPT_HEX_SUFFIXES_SHORT: &str = "-x";
 static OPT_SUFFIX_LENGTH: &str = "suffix-length";
 static OPT_VERBOSE: &str = "verbose";
 static OPT_SEPARATOR: &str = "separator";
-//The ---io and ---io-blksize parameters are consumed and ignored.
-//The parameter is included to make GNU coreutils tests pass.
-static OPT_IO: &str = "-io";
-static OPT_IO_BLKSIZE: &str = "-io-blksize";
 static OPT_ELIDE_EMPTY_FILES: &str = "elide-empty-files";
+static OPT_IO_BLKSIZE: &str = "-io-blksize";
+// Cap ---io-blksize value
+// For 64bit systems the max value is the same as in GNU
+// and is equivalent of `i32::MAX >> 20 << 20` operation.
+// On 32bit systems however, even though it fits within `u32` and `i32`,
+// it causes rust-lang `library/alloc/src/raw_vec.rs` to panic with 'capacity overflow' error.
+// Could be due to how `std::io::BufReader` handles internal buffers.
+// So we use much smaller value for those
+static OPT_IO_BLKSIZE_MAX: usize = if usize::BITS >= 64 {
+    2_146_435_072
+} else {
+    1_000_000_000
+};
+// The ---io parameter is consumed and ignored.
+// The parameter is included to make GNU coreutils tests pass.
+static OPT_IO: &str = "-io";
```

Comment:
Here's an example from GNU:
```
❯ split ---
split: option '---io-blksize' requires an argument
Try 'split --help' for more information.
❯ split ---io-bl
split: option '---io-blksize' requires an argument
Try 'split --help' for more information.
```

This is not a separate option, but one of those things where a long argument can be shortened to the first few letters if its unambiguous. `clap` supports this too and I think it's turned on (it's `infer_long_args(true)`) so do we need this?

Reply:
yep, looks like it is the case. will remove